mobify/generator-stencil
# Add schema template.

- Add template for the component schema based on [js-schema](https://github.com/molnarg/js-schema/);
- Add example data
- Add task-based validation of example data using the schema.

Fixes #5.

Backround: the schema is a description of the data required by a component’s Dust template. It should serve as both documentation for the team and potentially for validating the data being passed to a component in integration tests for a view. Schemas and example (mock) data are distinct.

Status: **Ready for review**

Reviewers: CS Eng team
Ticket: [CSOPS-1271](https://mobify.atlassian.net/browse/CSOPS-1271)
Linked PRs: none

## Changes
- Adds the js-schema library
- Updates the component schema to use the correct format for js-schema schemas
- Adds some inline docs to the schema file
- Adds example data that conforms to the schema
- Adds task-based validation of example data using the schema (uses mocha)

## How to test-drive this PR

Code review is what’s needed most with this PR. If you’d like to test the automated example data validation, do this:
- Clone this repo
- cd `generator-stencil`
- `npm link`
- cd `some/temp/directory`
- `yo stencil` and follow the prompts. Any input is fine. NPM and Bower install will be run automatically.
- cd into the new component directory. It will be prefixed with `stencil-`
- run `grunt test_schema`. There is only one test; it should pass.

## Research resources
- https://github.com/molnarg/js-schema/
- [Stencil: A Component’s Journey (working spec)](https://docs.google.com/a/mobify.me/document/d/1NlhC7nsLGORsMY9mtuB3FmO7VVNq2Mdse67SCBXUWho/edit#)
